tests/UnifiedSpecTests/UnifiedSpecTest.php

```diff
@@ -3,463 +3,123 @@
-            $this->prepareInitialData($initialData);
-        }
-
-        // Give Context unmodified URI so it can enforce useMultipleMongoses
-        $context = new Context(self::$internalClient, static::getUri(true));
-
-        if (isset($createEntities)) {
-            $context->createEntities($createEntities);
-        }
-
-        $this->assertIsArray($test->operations);
-        $this->preventStaleDbVersionError($test->operations, $context);
-
-        $context->startEventObservers();
-
-        foreach ($test->operations as $o) {
-            $operation = new Operation($o, $context);
-            $operation->assert();
-        }
-
-        $context->stopEventObservers();
-
-        if (isset($test->expectEvents)) {
-            $this->assertIsArray($test->expectEvents);
-            $context->assertExpectedEventsForClients($test->expectEvents);
-        }
-
-        if (isset($test->outcome)) {
-            $this->assertIsArray($test->outcome);
-            $this->assertOutcome($test->outcome);
-        }
+        return $this->provideTests(__DIR__ . '/change-streams/*.json');
     }
 
     /**
      * @dataProvider provideCrudTests
      */
-    public function testCrud(...$args)
+    public function testCrud(UnifiedTestCase $test)
     {
-        $this->doTestCase(...$args);
+        self::$runner->run($test);
     }
 
     public function provideCrudTests()
     {
-        return $this->provideTests(__DIR__ . '/crud');
+        return $this->provideTests(__DIR__ . '/crud/*.json');
     }
 
     /**
      * @dataProvider provideVersionedApiTests
      */
-    public function testVersionedApi(...$args)
+    public function testVersionedApi(UnifiedTestCase $test)
     {
-        $this->doTestCase(...$args);
+        self::$runner->run($test);
     }
 
     public function provideVersionedApiTests()
     {
-        return $this->provideTests(__DIR__ . '/versioned-api');
+        return $this->provideTests(__DIR__ . '/versioned-api/*.json');
     }
 
     /**
      * @dataProvider providePassingTests
      */
-    public function testPassingTests(...$args)
+    public function testPassingTests(UnifiedTestCase $test)
     {
-        $this->doTestCase(...$args);
+        self::$runner->run($test);
     }
 
     public function providePassingTests()
     {
-        return $this->provideTests(__DIR__ . '/valid-pass');
+        yield from $this->provideTests(__DIR__ . '/valid-pass/*.json');
     }
 
     /**
      * @dataProvider provideFailingTests
      */
-    public function testFailingTests(...$args)
+    public function testFailingTests(UnifiedTestCase $test)
     {
         // Cannot use expectException(), as it ignores PHPUnit Exceptions
         $failed = false;
 
         // phpcs:disable SlevomatCodingStandard.Exceptions.ReferenceThrowableOnly.ReferencedGeneralException
         try {
-            $this->doTestCase(...$args);
+            self::$runner->run($test);
         } catch (Exception $e) {
```

Comment:
Any particular reason why you're not catching `Throwable` here? Would that catch exceptions that we're not interested in?

Reply:
This is needed to ensure we don't consider PHP errors and is related to other changes in the "Rethrow internal PHP errors in Operation::assert()" comment.

I'll add a comment to explain this.